### PR TITLE
Make pylint happy

### PR DESCRIPTION
**Pull request by liuzhe-lz**
PPO tuner is not touched because I can't understand reported functions. @QuanluZhang 
I have not installed all tuner dependencies and disabled `import-error` locally, so it may be incomplete.

### CODE REVIEW

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 1]
```diff
@@ -191,22 +191,23 @@ def _add_index(in_x, parameter):
                     return {NodeType.INDEX: pos, NodeType.VALUE: item}
         else:
             return parameter
+    return None  # FIXME: what to return here?
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: @xuehui1991 Please check this.

File: src/sdk/pynni/nni/networkmorphism_tuner/layers.py [hunk 2]
```diff
@@ -576,6 +570,7 @@ def to_real_keras_layer(layer):
         return layers.Flatten()
     if is_layer(layer, "GlobalAveragePooling"):
         return layers.GlobalAveragePooling2D()
+    return None  # FIXME: should here return or raise?
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/networkmorphism_tuner/layers.py]: @xuehui1991 And this.

File: src/sdk/pynni/nni/networkmorphism_tuner/layers.py [hunk 3]
```diff
@@ -608,6 +603,7 @@ def is_layer(layer, layer_type):
         return isinstance(layer, (StubFlatten,))
     elif layer_type == "GlobalAveragePooling":
         return isinstance(layer, StubGlobalPooling)
+    return None  # FIXME: return or raise?
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/networkmorphism_tuner/layers.py]: And this

Comment (xuehui1991) [reply src/sdk/pynni/nni/networkmorphism_tuner/layers.py]: return None. do not change the logic, thank you.

File: src/sdk/pynni/nni/networkmorphism_tuner/layers.py (hunk 2 above)
Comment (xuehui1991) [reply src/sdk/pynni/nni/networkmorphism_tuner/layers.py]: do not change the logic.

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py (hunk 1 above)
Comment (xuehui1991) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: Do not change the logic in this PR.

Comment (liuzhe-lz) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: Okay. Though it's still smelly to me.
I'll change FIXME into a note.

File: src/sdk/pynni/nni/assessor.py [hunk 4]
```diff
@@ -46,21 +45,20 @@ def trial_end(self, trial_job_id, success):
         trial_job_id: identifier of the trial (str).
         success: True if the trial successfully completed; False if failed or terminated.
         """
-        pass
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/assessor.py]: @leckie-chn Is this recommended style?
This will cause `W0107: Unnecessary pass statement (unnecessary-pass)`

Comment (leckie-chn) [reply src/sdk/pynni/nni/assessor.py]: Currently in our `pylintrc` file `W1017` is not enabled. Are you sure you are using the right version of `pylintrc` file?

**Comment by leckie-chn** [comment]
@liuzhe-lz  Just to clarify, which modules is this PR fixing?

**Comment by leckie-chn** [comment]
With the output on nni SDK with current settings:
```
python3 -m pylint src/sdk/pynni/nni
************* Module nni.bohb_advisor.config_generator
src/sdk/pynni/nni/bohb_advisor/config_generator.py:258:12: E1137: 'return_array' does not support item assignment (unsupported-assignment-operation)************* Module nni.compression.tensorflow.builtin_quantizers
src/sdk/pynni/nni/compression/tensorflow/builtin_quantizers.py:30:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)src/sdk/pynni/nni/compression/tensorflow/builtin_quantizers.py:55:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)************* Module nni.compression.tensorflow
src/sdk/pynni/nni/compression/tensorflow/__init__.py:2:0: W0401: Wildcard import builtin_pruners (wildcard-import)
src/sdk/pynni/nni/compression/tensorflow/__init__.py:3:0: W0401: Wildcard import builtin_quantizers (wildcard-import)
************* Module nni.compression.torch.builtin_quantizers
src/sdk/pynni/nni/compression/torch/builtin_quantizers.py:30:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/compression/torch/builtin_quantizers.py:56:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
************* Module nni.compression.torch
src/sdk/pynni/nni/compression/torch/__init__.py:2:0: W0401: Wildcard import builtin_pruners (wildcard-import)
src/sdk/pynni/nni/compression/torch/__init__.py:3:0: W0401: Wildcard import builtin_quantizers (wildcard-import)
************* Module nni.curvefitting_assessor.curvefunctions
src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py:103:4: E1111: Assigning to function call which doesn't return (assignment-from-no-return)
************* Module nni.curvefitting_assessor.model_factory
src/sdk/pynni/nni/curvefitting_assessor/model_factory.py:311:12: E1111: Assigning to function call which doesn't return (assignment-from-no-return)
************* Module nni.evolution_tuner.test_evolution_tuner
src/sdk/pynni/nni/evolution_tuner/test_evolution_tuner.py:54:32: C0326: Exactly one space required after comma
                "_type":"choice","_value":["Adam", "SGD"]
                                ^ (bad-whitespace)
src/sdk/pynni/nni/evolution_tuner/test_evolution_tuner.py:26:0: C0411: standard import "from unittest import TestCase, main" should be placed before "import numpy as np" (wrong-import-order)
************* Module nni.gp_tuner.target_space
src/sdk/pynni/nni/gp_tuner/target_space.py:142:61: E1102: self.dim is not callable (not-callable)
************* Module nni.hyperopt_tuner.hyperopt_tuner
src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py:59:20: E1111: Assigning to function call which doesn't return (assignment-from-no-return)
************* Module nni.networkmorphism_tuner.graph
src/sdk/pynni/nni/networkmorphism_tuner/graph.py:801:8: E0401: Unable to import 'keras' (import-error)
src/sdk/pynni/nni/networkmorphism_tuner/graph.py:905:4: E0401: Unable to import 'onnx' (import-error)
************* Module nni.networkmorphism_tuner.layers
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:33:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:412:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:471:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:528:4: E0401: Unable to import 'keras' (import-error)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:544:4: E0401: Unable to import 'keras' (import-error)
************* Module nni.networkmorphism_tuner.test_networkmorphism_tuner
src/sdk/pynni/nni/networkmorphism_tuner/test_networkmorphism_tuner.py:195:27: C0326: Exactly one space required after comma
        tuner.add_model(0.8,  0)
                           ^ (bad-whitespace)
src/sdk/pynni/nni/networkmorphism_tuner/test_networkmorphism_tuner.py:196:27: C0326: Exactly one space required after comma
        tuner.add_model(0.9,  1)
                           ^ (bad-whitespace)
************* Module nni.ppo_tuner.distri
src/sdk/pynni/nni/ppo_tuner/distri.py:153:15: E1120: No value for argument 'mask_npinf' in constructor call (no-value-for-parameter)
src/sdk/pynni/nni/ppo_tuner/distri.py:153:15: E1120: No value for argument 'nsteps' in constructor call (no-value-for-parameter)
src/sdk/pynni/nni/ppo_tuner/distri.py:153:15: E1120: No value for argument 'size' in constructor call (no-value-for-parameter)
src/sdk/pynni/nni/ppo_tuner/distri.py:153:15: E1120: No value for argument 'is_act_model' in constructor call (no-value-for-parameter)
************* Module nni.ppo_tuner.ppo_tuner
src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py:540:0: C0330: Wrong continued indentation (remove 2 spaces).
                                        mb_dones, last_values, self.inf_batch_size)
                                      | ^ (bad-continuation)
src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py:239:12: E1137: 'mb_advs' does not support item assignment (unsupported-assignment-operation)
src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py:584:27: W1202: Use % formatting in logging functions and pass the % parameters as arguments (logging-format-interpolation)
src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py:585:80: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py:25:0: W0611: Unused import os (unused-import)
src/sdk/pynni/nni/ppo_tuner/ppo_tuner.py:29:0: W0611: Unused import json_tricks (unused-import)
************* Module nni.ppo_tuner.util
src/sdk/pynni/nni/ppo_tuner/util.py:59:4: W0612: Unused variable 'nbatch' (unused-variable)
************* Module nni.smac_tuner.smac_tuner
src/sdk/pynni/nni/smac_tuner/smac_tuner.py:282:20: E1111: Assigning to function call which doesn't return (assignment-from-no-return)

------------------------------------------------------------------
Your code has been rated at 9.85/10 (previous run: 9.39/10, +0.46)
```

File: src/sdk/pynni/nni/assessor.py (hunk 4 above)
Comment (liuzhe-lz) [reply src/sdk/pynni/nni/assessor.py]: I'm using the rc file included in this PR.

**Comment by liuzhe-lz** [comment]
@leckie-chn For your pylintrc output:
* PPO tuner is not covered in this PR
* Unit tests should not be required to pass pylint
* I disabled `useless-super-delegation` until we find a way to write derived constructor docstring
* `unsupported-assignment-operation` and `assignment-from-no-return` are because numpy not marked as `generated-members` in rc file
* The other three files strangely passed pylint on my machine and will be updated soon

**Comment by leckie-chn** [comment]
> @leckie-chn For your pylintrc output:
> 
> * PPO tuner is not covered in this PR
> * Unit tests should not be required to pass pylint
> * I disabled `useless-super-delegation` until we find a way to write derived constructor docstring
> * `unsupported-assignment-operation` and `assignment-from-no-return` are because numpy not marked as `generated-members` in rc file
> * The other three files strangely passed pylint on my machine and will be updated soon

@liuzhe-lz https://github.com/microsoft/nni/blob/041c3f0e2d1f0020f8c30e422094a07bb88b9b2d/pylintrc#L45 Here we have a rule for disabling check for Unit tests.

**Comment by liuzhe-lz** [comment]
@leckie-chn Suggest use `ignore-patterns`.

File: src/sdk/pynni/nni/assessor.py (hunk 4 above)
Comment (leckie-chn) [reply src/sdk/pynni/nni/assessor.py]: You should use https://github.com/microsoft/nni/blob/master/pylintrc

**Comment by leckie-chn** [comment]
> @leckie-chn For your pylintrc output:
> 
> * PPO tuner is not covered in this PR
> * Unit tests are not required to pass pylint (though recommended maybe)
> * I disabled `useless-super-delegation` until we find a way to write derived constructor docstring
> * `unsupported-assignment-operation` and `assignment-from-no-return` are because numpy not marked as `generated-members` in rc file
> * `wildcard-import` is disabled for `__init__.py` in latest pylint, [see here](https://github.com/PyCQA/pylint/pull/2084)
> * The last problem in GP tuner is strangely ignored by latest pylint (fixed below)

output w/ `pylint 2.4.3`, removing all UTs and PPO modules:
```
************* Module nni.bohb_advisor.config_generator
src/sdk/pynni/nni/bohb_advisor/config_generator.py:312:69: E1136: Value 'train_configs.shape' is unsubscriptable (unsubscriptable-object)
src/sdk/pynni/nni/bohb_advisor/config_generator.py:313:72: E1136: Value 'train_configs.shape' is unsubscriptable (unsubscriptable-object)
src/sdk/pynni/nni/bohb_advisor/config_generator.py:321:11: E1136: Value 'train_data_good.shape' is unsubscriptable (unsubscriptable-object)
src/sdk/pynni/nni/bohb_advisor/config_generator.py:321:39: E1136: Value 'train_data_good.shape' is unsubscriptable (unsubscriptable-object)
src/sdk/pynni/nni/bohb_advisor/config_generator.py:323:11: E1136: Value 'train_data_bad.shape' is unsubscriptable (unsubscriptable-object)
src/sdk/pynni/nni/bohb_advisor/config_generator.py:323:38: E1136: Value 'train_data_bad.shape' is unsubscriptable (unsubscriptable-object)
************* Module nni.compression.tensorflow.builtin_quantizers
src/sdk/pynni/nni/compression/tensorflow/builtin_quantizers.py:30:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)src/sdk/pynni/nni/compression/tensorflow/builtin_quantizers.py:55:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)************* Module nni.compression.torch.builtin_quantizers
src/sdk/pynni/nni/compression/torch/builtin_quantizers.py:30:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)     src/sdk/pynni/nni/compression/torch/builtin_quantizers.py:56:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)     ************* Module nni.metis_tuner.metis_tuner
src/sdk/pynni/nni/metis_tuner/metis_tuner.py:358:43: E1133: Non-iterable value results_outliers is used in an iterating context (not-an-iterable)   ************* Module nni.networkmorphism_tuner.graph
src/sdk/pynni/nni/networkmorphism_tuner/graph.py:801:8: E0401: Unable to import 'keras' (import-error)
src/sdk/pynni/nni/networkmorphism_tuner/graph.py:905:4: E0401: Unable to import 'onnx' (import-error)
************* Module nni.networkmorphism_tuner.layers
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:33:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:412:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:471:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:528:4: E0401: Unable to import 'keras' (import-error)
src/sdk/pynni/nni/networkmorphism_tuner/layers.py:544:4: E0401: Unable to import 'keras' (import-error)
```

* `unsubscriptable-object` is caused by https://github.com/PyCQA/pylint/issues/3139 
* For disabling `useless-super-delegation` if you want to locally disable it, please add the comment `#pylint: disable=useless-super-delegation`.
* `not-an-iterable` is the 1 true real warning that needs to be fixed.

**Comment by liuzhe-lz** [comment]
@leckie-chn 
* See our old pylintrc file. `generated-members=numpy` can stop pylint to analyze numpy types.
* I believe it's a good practice to refine parent class' docstring. And I don't want to make good things verbose. If a class has a useless `__init__` without docstring, it will be later detected by `missing-docstring`.
* It looks like an iterable to me. `results_outliers` is returned by `gp_oulier_detection.outlierDetection_threaded`, which is defined in `nni/metis_tuner/Regression_GP/OutlierDetection.py`. The function returns either a list or `None`, and there is a `is not None` checking. So this should be a pylint bug.

**Pull request merged**